contracts/src/fp/OPSuccinctFaultDisputeGame.sol

```diff
@@ -407,51 +477,50 @@ contract OPSuccinctFaultDisputeGame is Clone, ISemver {
             status = GameStatus.CHALLENGER_WINS;
             resolvedAt = Timestamp.wrap(uint64(block.timestamp));
 
-            // Record the challenger's reward
-            credit[claimData.counteredBy] += address(this).balance;
+            // Add challenger's reward to the entry point contract.
+            (bool success,) = ENTRY_POINT.call{value: address(this).balance}(
+                abi.encodeWithSelector(
+                    OPSuccinctEntryPoint.addCredit.selector, claimData.counteredBy, address(this).balance
+                )
+            );
+            if (!success) revert BondTransferFailed();
 
             emit Resolved(status);
         } else if (claimData.status == ProposalStatus.UnchallengedAndValidProofProvided) {
             claimData.status = ProposalStatus.Resolved;
```

Comment:
I'd add a clarifying comment here about the fact that when the game state is unchallenged, there's no reward for a third-party prover. The proposer just gets re-funded the bond. This prevents front-running attacks on the proposer.